SwiftCompilerSources/CMakeLists.txt

```diff
@@ -231,6 +232,16 @@ function(add_swift_compiler_modules_library name)
   add_library(${name} STATIC ${all_obj_files})
   add_dependencies(${name} ${all_module_targets})
   set_target_properties(${name} PROPERTIES LINKER_LANGUAGE CXX)
+
+  if(SWIFT_HOST_VARIANT_SDK IN_LIST SWIFT_DARWIN_PLATFORMS)
```

Comment:
Not sure if there's a better check that can be used here to test for the linker being able to support `-add_ast_path`, this is the check that guarded the previous logic though.